subsys/bluetooth/host/iso.c

```diff
@@ -777,9 +779,33 @@ static void bt_iso_remove_data_path(struct bt_conn *conn)
 {
 	BT_DBG("%p", conn);
 
-	/* Remove both directions */
-	hci_le_remove_iso_data_path(conn, BT_HCI_DATAPATH_DIR_CTLR_TO_HOST);
-	hci_le_remove_iso_data_path(conn, BT_HCI_DATAPATH_DIR_HOST_TO_CTLR);
+	if (conn->iso.is_bis) {
+		struct bt_iso_chan *chan;
+		struct bt_iso_chan_io_qos *tx_qos;
+		uint8_t dir;
+
+		chan = SYS_SLIST_PEEK_HEAD_CONTAINER(&conn->channels, chan, node);
```

Comment:
Usually the remove_data_path is only called when all channels have been removed, or this is not the case for BIS? Anyway it might be better that we store the direction(s) within conn->iso then to avoid such lookups so both BIS and CIS should become pretty similar if we can tell what data path have been programmed.

Reply:
For BIS they act more as a group: All BIS are either created or removed at the same time (i.e. you create or terminate a BIG (or a BIG sync), rather than individual BISes).

Reply:
@Vudentz Just did a test for 2 BIS in a BIG and this works as intended :)